packages/server/graphql/queries/tasks.ts

```diff
@@ -18,40 +18,61 @@ export default {
       type: GraphQLISO8601Type,
       description: 'the datetime cursor'
     },
-    teamId: {
-      type: GraphQLID,
-      description: 'The unique team ID'
+    userIds: {
+      type: GraphQLList(GraphQLNonNull(GraphQLID)),
+      description: 'a list of user Ids'
+    },
+    teamIds: {
+      type: GraphQLList(GraphQLNonNull(GraphQLID)),
+      description: 'a list of team Ids'
     },
     archived: {
       type: GraphQLBoolean,
-      description: 'true if only archived tasks are returned; false otherwise',
+      description: 'true to only return archived tasks; false to return active tasks',
       defaultValue: false
-    },
-    includeTeamMembers: {
-      type: GraphQLBoolean,
-      description: "true if tasks from user's team members are returned; false otherwise",
-      defaultValue: true
     }
   },
   async resolve(
-    _source,
-    {first, after, teamId, archived, includeTeamMembers},
+    {tms},
+    {first, after, userIds, teamIds, archived},
     {authToken, dataLoader}: GQLContext
   ) {
     // AUTH
     const viewerId = getUserId(authToken)
-    if (teamId && !isTeamMember(authToken, teamId)) {
-      standardError(new Error('Team not found'), {userId: viewerId})
-      return connectionFromTasks([])
+    if (teamIds) {
+      for (const teamId of teamIds) {
+        // cannot query tasks from teams the viewer is not in
+        if (!isTeamMember(authToken, teamId)) {
+          standardError(new Error('Team not found'), {userId: viewerId})
+          return connectionFromTasks([])
+        }
+      }
     }
-    const teamIds = teamId ? [teamId] : authToken.tms || []
+
+    let userIdsForQuery = userIds
+    let teamIdsForQuery = teamIds
+
+    if (!userIds && !teamIds) {
+      userIdsForQuery = [viewerId]
+      teamIdsForQuery = tms
+    } else if (userIds && !teamIds) {
+      const users = await dataLoader.get('users').loadMany(userIds)
```

Comment:
this will error if `userIds === []`

one possible solution is making `teamIds` and `userIds` nonnull inputs to prevent junk inputs from untrusted clients.